tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+  else
+  begin
+    // using ImageMagic - FPWriteXPM first doesn't properly write alpha channel, second uses palette char size = 2 which is not a good idea for an icon
+    //RunCommandSimple(FindExe('convert'), [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal]);
+    if not RunCommand('/bin/convert', [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal], OutString) then
+      WriteLn('ImageMagick failed.');
+    WriteLn(OutString);
+  end;
+
+  // Create menu item for the game
+
+  CreateDir(ShareDir + PathDelim + 'menu');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'menu' + PathDelim + Manifest.Name);
+  TextWriter.Write(
+    '?package(' + Manifest.Name + '): \' + NL +
+    'needs="X11" \' + NL +
+    'section="' + Manifest.DebianSection + '" \' + NL +
+    'title="' + Manifest.Caption + '" \' + NL +
+    'command="bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''" \' + NL +
+    'icon="' + PathToIconFileUnix +'"'
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(ShareDir + PathDelim + 'applications');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'applications' + PathDelim + Manifest.ExecutableName + '.desktop');
+  TextWriter.Write(
+    '[Desktop Entry]' + NL +
+    'Version=' + Manifest.Version.DisplayValue + NL +
+    'Terminal=false' + NL +
+    'Exec=bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''' + NL +
+    'Icon=' + PathToIconFileUnix + NL +
+    'Type=Application' + NL +
+    'Categories=' + Manifest.DebianCategories + NL +
+    'Name=' + Manifest.Caption + NL +
+    'Comment=' + Manifest.DebianComment
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(PackageFolder + PathDelim + 'DEBIAN');
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'control');
+  TextWriter.Write(
+    'Package: ' + Manifest.Name + NL +
+    'Version: ' + Manifest.Version.DisplayValue + NL +
+    'Section: ' + Manifest.DebianInstallFolder + NL +
+    'Priority: optional' + NL +
+    'Architecture: ' + CpuToArchitectureString(Cpu) + NL +
+    'Maintainer: ' + Manifest.Author + NL +
+    'Installed-Size: ' + IntToStr(BinariesSize div 1024) + NL +
+    'Depends: libopenal1, libpng16-16, zlib1g, libvorbis0a, libvorbisfile3, libfreetype6, libgl1-mesa-dri, libgtk2.0-0' + NL +
+    'Description: ' + Manifest.Caption + NL +
+    ' ' + Manifest.DebianComment + NL //final new line
+    );
+  FreeAndNil(TextWriter);
+
+  // Post-installation running - assign executable permissions
+
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst');
+  TextWriter.Write(
+    '#!/bin/sh' + NL + NL +
+    'chmod +x ' + PathToExecutableUnix + '/' + Manifest.ExecutableName + NL + NL +
+    'exit 0'
+    );
+  FreeAndNil(TextWriter);
+  fpChmod(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst', &555);
```

Comment:
Jenkins reports a compilation error here, and is right ( https://jenkins.castle-engine.io/job/castle_game_engine_organization/job/castle-engine/job/PR-302/5/console  ):

```
... for Win64:
...
Free Pascal Compiler version 3.2.0 [2020/06/14] for x86_64
Copyright (c) 1993-2020 by Florian Klaempfl and others
tooldebian.pas(168,3) Error: Identifier not found "fpChmod"
```

Indeed `FpChmod` should be under `{$ifdef UNIX}`.

Actually, make it even better: move the `DoMakeExecutable` from `toolpackage.pas` into `toolutils.pas`. 

Then
- `TPackageDirectory.Add` can call `DoMakeExecutable(Path + Name)`
- and your new code can call `DoMakeExecutable(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst')`
- and the `$ifdef` is hidden inside `DoMakeExecutable`.

Reply:
Done in 888690fdac181b6f140a71fd0d5ac20a7d7b59e6